src/sage/matroids/flats_matroid.pxd

```diff
@@ -0,0 +1,20 @@
+from sage.matroids.matroid cimport Matroid
+
+cdef class FlatsMatroid(Matroid):
+    cdef frozenset _groundset  # _E
+    cdef int _matroid_rank  # _R
+    cdef dict _F  # flats
+    cpdef groundset(self) noexcept
```

Comment:
note that over in #37667, we are getting rid of many `noexcept` directives that had been placed incorrectly. Basically anything with `self` or involving any other Python objects shouldn't be `noexcept`

Reply:
Great, happy to be getting rid of noisy stuff.